src/main/java/net/fabricmc/loader/api/metadata/version/VersionComparisonOperator.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.api.metadata.version;
+
+import net.fabricmc.loader.api.SemanticVersion;
+import net.fabricmc.loader.api.Version;
+import net.fabricmc.loader.impl.util.version.SemanticVersionImpl;
+
+public enum VersionComparisonOperator {
+	// order is important to match the longest substring (e.g. try >= before >)
+	GREATER_EQUAL(">=", true, false) {
+		@Override
+		public boolean test(SemanticVersion a, SemanticVersion b) {
+			return a.compareTo((Version) b) >= 0;
+		}
+
+		@Override
+		public SemanticVersion minVersion(SemanticVersion version) {
+			return version;
+		}
+	},
+	LESS_EQUAL("<=", false, true) {
+		@Override
+		public boolean test(SemanticVersion a, SemanticVersion b) {
+			return a.compareTo((Version) b) <= 0;
+		}
+
+		@Override
+		public SemanticVersion maxVersion(SemanticVersion version) {
+			return version;
+		}
+	},
+	GREATER(">", false, false) {
+		@Override
+		public boolean test(SemanticVersion a, SemanticVersion b) {
+			return a.compareTo((Version) b) > 0;
+		}
+
+		@Override
+		public SemanticVersion minVersion(SemanticVersion version) {
+			return version;
+		}
+	},
+	LESS("<", false, false) {
+		@Override
+		public boolean test(SemanticVersion a, SemanticVersion b) {
+			return a.compareTo((Version) b) < 0;
+		}
+
+		@Override
+		public SemanticVersion maxVersion(SemanticVersion version) {
+			return version;
+		}
+	},
+	EQUAL("=", true, true) {
+		@Override
+		public boolean test(SemanticVersion a, SemanticVersion b) {
+			return a.compareTo((Version) b) == 0;
+		}
+
+		@Override
+		public SemanticVersion minVersion(SemanticVersion version) {
+			return version;
+		}
+
+		@Override
+		public SemanticVersion maxVersion(SemanticVersion version) {
+			return version;
+		}
+	},
+	SAME_TO_MEXT_MINOR("~", true, false) {
```

Comment:
Typo of `NEXT`

Reply:
good catch, that cursive monospace font in Eclipse really doesn't help there :D